Source/diablo.cpp

```diff
@@ -1649,26 +1662,27 @@ void LoadGameLevel(BOOL firstflag, int lvldir)
 		if (leveltype != DTYPE_TOWN) {
 			if (firstflag || lvldir == ENTRY_LOAD || !plr[myplr]._pLvlVisited[currlevel] || gbMaxPlayers != 1) {
 				HoldThemeRooms();
-				glMid1Seed[currlevel] = GetRndSeed();
+				glMid1Seed[currlevel] = sglGameSeed;

```

Comment:
I don't think I get why this is needed?

Reply:
because GetRndSee() modifies the seed?

Reply:
It can screw up the seed. IMO best way to handle it to have to RNG's one strictly for dungeon generation and another for the rest that can be called without worry.

Reply:
this was the only rng desync I've spotted so it works fine in general I'd say except this where they did not one but two mistakes :D